.github/workflows/vfd-subfiling.yml

```diff
@@ -0,0 +1,136 @@
+name: Test HDF5 Subfiling VFD
+
+on:
+  workflow_call:
+    inputs:
+      build_sys:
+        description: "build system used"
+        required: true
+        type: string      
+      build_mode:
+        description: "release vs. debug build"
+        required: true
+        type: string
+
+permissions:
+  contents: read
+
+jobs:
+  build_and_test:
+    strategy:
+      # Let jobs run to completion even if one fails
+      fail-fast: false
+      matrix:
+        os_name: ["Ubuntu"]
+#        os_name: ["Ubuntu", "MacOS"]
```

Comment:
Commented out somethings here that need more work. MacOS testing needs added; MPICH testing will require building from source since the default version from APT on Ubuntu hangs almost immediately in Subfiling. I'd like to create separate actions for building and then caching OpenMPI and MPICH so that all of our actions can reuse that rather than constantly rebuilding it or installing a likely old version from apt